File: padroesComportamentais/padrao-de-projeto-interpreter/php/tests/ClienteTest.php
<?php

namespace interpreter\tests;

use Interpreter\Php\Cliente;
use PHPUnit\Framework\TestCase;

class ClienteTest extends TestCase
{
    /** @test */
    public function deveCalcularPrecoProduto(): void
    {
        $cliente = new Cliente();
        $this->assertEquals(
            60,
            $cliente->importaProduto(10, 20)
        );
    }
}

File: padroesComportamentais/padrao-de-projeto-interpreter/java/src/test/java/ClienteTest.java
import org.junit.Test;

import static org.junit.jupiter.api.Assertions.*;

public class ClienteTest {

    @Test
    public void deveCalcularPrecoProduto() {
        Cliente cliente = new Cliente();
        Double precoProdutoImportado = cliente.importaProduto(15, 10);
        assertEquals(50, precoProdutoImportado);
    }
}
